distribution/push_v2.go

```diff
@@ -169,7 +170,13 @@ func (p *v2Pusher) pushV2Tag(ctx context.Context, ref reference.NamedTagged, ima
 
 	putOptions := []distribution.ManifestServiceOption{distribution.WithTag(ref.Tag())}
 	if _, err = manSvc.Put(ctx, manifest, putOptions...); err != nil {
-		logrus.Warnf("failed to upload schema2 manifest: %v - falling back to schema1", err)
+		logrus.Warnf("failed to upload schema2 manifest: %v", err)
+
+		if runtime.GOOS == "windows" {
+			return err
+		}
```

Comment:
We don't expect to be compatible with older registries at all, since they won't support the correct handling of base layers for Windows images.